docs/core/session.md

```diff
@@ -2,3 +2,53 @@
 index: 2
 title: "Session"
 ---
+
+Leapkit provides a way to set a session in your project to persist values needed throughout the project. Leapkit uses the [gorilla/sessions](https://github.com/gorilla/sessions) package for this.
+
+## Setup
+
+To set the session in your project, you have to use the `server.WithSession` server option and pass a session secret key and the session name.
+
+```go
+s := server.New(
+   server.WithSession("secret_key", "session_name"),
+)
+```
+
+## Handling session values and flashes
+
+To use the session struct within your handler, retrieve it from the context using the `session.FromCtx()` function. Then, you can manage your session values according to the `gorilla/session` package [docs](https://pkg.go.dev/github.com/gorilla/sessions). For instance:
+
+
+```go
+func Handler(w http.ResponseWriter, r *http.Request) {
+    ss := session.FromCtx(r.Context())
+
+    // setting a session value
+    ss.Values["sessionVal"] = "value"
+
+    // getting a session value
+    myVal := ss.Values["sessionVal"].(string)
+
+
+    // setting a flash message
+    ss.AddFlash("welcome!")
+    ss.AddFlash("peter", "username_flash")
+    // {
+    // 	"_flash": ["welcome!"],
+    // 	"username_flash": ["peter"],
+    // }
+
+    // getting a flash messages
+    flashes := ss.Flashes("username_flash")
+    // ["peter"]
+    // ...
+
+    // Saving the session
+    if err := ss.Save(r, w); err != nil {
+        // handle the error
+    }
```

Comment:
This part looks confusing to me @tatang26, can we avoid the Save?

Reply:
Sure, I'll remove it